api/v1/model/src/commonMain/kotlin/OrtRun.kt

```diff
@@ -160,7 +160,8 @@ enum class OrtRunStatus {
     CREATED,
     ACTIVE,
     FINISHED,
-    FAILED
+    FAILED,
```

Comment:
As a side note, having e.g. test execution statuses in mind, I find "failed" already confusing here. For tests, it means that the test *did* run but the assertions failed. Whereas "error" means that a test did not run.

I'd actually propose to use the same terminology here:
- `FAILED` -> `ERROR`
- Introduce a new `FAILED` or `FAILED_WITH_ISSUES`.

Reply:
I don't think test terminology is a good fit here, it is better to look at other pipeline implementations. For example:

* Jenkins: SUCCESS, UNSTABLE, FAILURE, NOT_BUILT or ABORTED
* GitHub Workflows: completed, action_required, cancelled, failure, neutral, skipped, stale, success, timed_out, in_progress, queued, requested, waiting, pending 
* Azure Pipelines: Canceled, Failed, Succeeded, SucceededWithIssues, Skipped

What they have in common is the use of "success/succeeded" for a completely successful run and "failure/failed" for a completely failed run. This matches with our current terminology.

For the new status introduced in this commit Jenkins has "UNSTABLE" and Azure Pipelines has "SucceededWithIssues". The problem is to find a name that is not overloaded:
* "ERROR" is also the name of a severity, so does `FINISHED_WITH_ERRORS` mean that an issue or rule violation had `ERROR` severity? Actually it doesn't, because the threshold should be configurable.
* "ISSUES" is also used for the `Issue` class, so does `FINISHED_WITH_ISSUES` mean that there are issues with any severity? Does it include rule violations or vulnerabilities?

A not overloaded name could be `FINISHED_WITH_PROBLEMS`.

Reply:
@sschuberth  are you fine with `FINISHED_WITH_PROBLEMS` ?

Reply:
> it is better to look at other pipeline implementations.

Agreed.

> What they have in common is the use of "success/succeeded" for a completely successful run and "failure/failed" for a completely failed run. This matches with our current terminology.

Good.

> @sschuberth are you fine with `FINISHED_WITH_PROBLEMS` ?

I'm not sure yet. While I agree that the term "problems" is not overloaded / not used yet in ORT, it somehow sounds a bit too strong like, well, a problem, in the sense of something bad. But actually, e.g. for rule violations that might turn out to be false-positives, we don't know yet if really something bad happened.

What all of `Issue`, `RuleViolation` and `VulnerabilityReference` have in common is a `severity` property. So maybe call this `FINISHED_WITH_SEVERITY`? I'm not fully convinced yet myself, I'm just throwing in ideas.

Reply:
After thinking about the options my preferred name would be `FINISHED_WITH_ISSUES`. Even if it could be confused with the issues class the name itself is still very clear: the run could finished, but it found some issues.
`FAILED` should be reserved for runs that could not create a result.

Reply:
Yes sorry I meant to type `FINISHED_WITH_ISSUES` to :sweat_smile:

Reply:
Either `FINISHED_WITH_ISSUES` or `FINISHED_WITH_SEVERITY` are fine with me. Please remember the align the class names in this PR accordingly.